Examples/DrawSuite/drawsuite.cpp

```diff
@@ -0,0 +1,464 @@
+
+  // define the output clip
+  gEffectHost->clipDefine(effect, kOfxImageEffectOutputClipName, &props);
+  gPropHost->propSetString(props, kOfxImageEffectPropSupportedComponents, 0, kOfxImageComponentRGBA);
+  gPropHost->propSetString(props, kOfxImageEffectPropSupportedComponents, 1, kOfxImageComponentAlpha);
+  
+  // fetch the parameter set from the effect
+  OfxParamSetHandle paramSet;
+  gEffectHost->getParamSet(effect, &paramSet);
+
+  // define the 2D point we are going to draw an overlay for
+  gParamHost->paramDefine(paramSet, kOfxParamTypeDouble2D, kPointParam, &props);
+  gPropHost->propSetString(props, kOfxParamPropDoubleType, 0, kOfxParamDoubleTypeXYAbsolute);
+  gPropHost->propSetString(props, kOfxParamPropDefaultCoordinateSystem, 0, kOfxParamCoordinatesNormalised);
+  gPropHost->propSetDouble(props, kOfxParamPropDefault, 0, 0.5);
+  gPropHost->propSetDouble(props, kOfxParamPropDefault, 1, 0.5);
+  gPropHost->propSetString(props, kOfxParamPropHint, 0, "Point attached to overlay crosshair");
+  gPropHost->propSetString(props, kOfxParamPropScriptName, 0, "point");
+  gPropHost->propSetString(props, kOfxPropLabel, 0, "Point");
+
+  return kOfxStatOK;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// The main function
+static OfxStatus
+pluginMain(const char *action, const void *handle, OfxPropertySetHandle inArgs, OfxPropertySetHandle outArgs)
+{
+  try {
+  // cast to appropriate type
+  OfxImageEffectHandle effect = (OfxImageEffectHandle ) handle;
+
+  if(strcmp(action, kOfxActionDescribe) == 0) {
+    return describe(effect);
+  }
+  else if(strcmp(action, kOfxImageEffectActionDescribeInContext) == 0) {
+    return describeInContext(effect, inArgs);
+  }
+  else if(strcmp(action, kOfxImageEffectActionIsIdentity) == 0) {
+    return isIdentity(effect, inArgs, outArgs);
+  }    
+  else if(strcmp(action, kOfxImageEffectActionRender) == 0) {
+    return render(effect, inArgs, outArgs);
+  }    
+  } catch (std::bad_alloc) {
+    // catch memory
+    //std::cout << "OFX Plugin Memory error." << std::endl;
+    return kOfxStatErrMemory;
+  } catch ( const std::exception& e ) {
+    // standard exceptions
+    //std::cout << "OFX Plugin error: " << e.what() << std::endl;
+    return kOfxStatErrUnknown;
+  } catch (int err) {
+    // ho hum, gone wrong somehow
+    return err;
+  } catch ( ... ) {
+    // everything else
+    //std::cout << "OFX Plugin error" << std::endl;
+    return kOfxStatErrUnknown;
+  }
+    
+  // other actions to take the default value
+  return kOfxStatReplyDefault;
+}
+
+// function to set the host structure
+static void
+setHostFunc(OfxHost *hostStruct)
+{
+  gHost         = hostStruct;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// the plugin struct 
+static OfxPlugin basicPlugin = 
+{       
+  kOfxImageEffectPluginApi,
+  1,
+  "uk.co.thefoundry.BasicDrawSuitePlugin",
```

Comment:
Maybe this should be something like `org.openeffects.samples.DrawSuiteExample`?

Reply:
Yes you can change, this example predates the Foundry transfer